openedx/adg/admin_dashboard/webinars/views.py

```diff
@@ -93,3 +97,41 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+
+class WebinarCreateView(EnsureAdminUserAuthenticatedMixin, CreateView):
+    """
+    View to create a new webinar
+    """
+
+    model = Webinar
+    form_class = WebinarForm
+    template_name = 'adg/admin_dashboard/webinars/add_webinar.html'
+    success_url = reverse_lazy('webinars_list')
+
+    def get_context_data(self, **kwargs):
+        """
+        Override to format the form data as per the template fields
+        """
+        context = super(WebinarCreateView, self).get_context_data(**kwargs)
+        context['formatted_data'] = get_webinar_schedule_context_data(context['form'].data)
+        return context
+
+
+class WebinarUpdateView(EnsureAdminUserAuthenticatedMixin, UpdateView):
+    """
+    View to update an existing webinar
+    """
+
+    model = Webinar
+    form_class = WebinarForm
+    template_name = 'adg/admin_dashboard/webinars/update_webinar.html'
+    success_url = reverse_lazy('webinars_list')
+
+    def get_context_data(self, **kwargs):
+        """
+        Override to format the form data as per the template fields
+        """
+        context = super(WebinarUpdateView, self).get_context_data(**kwargs)
+        context['formatted_data'] = get_webinar_schedule_context_data(context['form'].data)
+        return context
```

Comment:
How about creating a mixin with this `get_context_data` and all the other common info, and then inheriting that class for both Update and Create Views.
It will remove the duplication as well.